### DISCUSSION TITLE
SyncCrazyflie vs Crazyflie

### DISCUSSION THREAD

**Discussion by NageshEranki** [bitcraze/discussions#1261] [Q&A] [answered]
Hi

I'm trying to understand the difference between the two methods of creating a Crazyflie instance:

```SyncCrazyflie()``` and ```Crazyflie()```. Online documentation reveals that ```SyncCrazyflie()``` is a wrapper to 
'handle the asynchronous nature of the API' and that it is well suited for tasks that must run in a sequence. 

If, for example,  one were to break down all tasks into three: logging, commanding set-points and sending pose measurements, is there a clear preference of one method over another? Will using  ```Crazyflie()``` provide me more freedom in controlling which tasks are running when?

Your example [script](https://github.com/bitcraze/crazyflie-lib-python/blob/master/examples/mocap/qualisys_hl_commander.py) is a nice reference for my objective however, I can't understand why ```SyncCrazyflie()``` is used here. I'm guessing that it is because the only 'task' is to send pose measurements to the CF. What difference would it make had I used ```Crazyflie()``` instead? 

Thanks
Nagesh

**Comment by ataffanel** [comment, marked as answer]
SyncCrazyfle exposes a subset of the Crazyflie API as easier to use synchronous functions. Crazyflie is the object that can handle everything but using asynchronous calls with callback function. SyncCrazyflie was designed to make simple task simple to implement.

So, if you are doing a simple script that fits the SyncCrazyflie function, you can use it. However, if you are doing more complex things you might need to use the Crazyflie either taking is from your SyncCrazyflie or instanciating a Crazyflie directly.

Said another way: the Crazyflie object is the underlying driver, so it can do everything SyncCrazyflie can but will require some more boilerplate code on the user side to do so.

I hope I was able to bring some clarity to your question.
